src/panels/config/integrations/integration-panels/zwave_js/add-node/zwave-js-add-node-code-input.ts

```diff
@@ -0,0 +1,99 @@
+import { customElement, property } from "lit/decorators";
+import { css, html, LitElement, nothing } from "lit";
+
+import { fireEvent } from "../../../../../../common/dom/fire_event";
+import type { HaTextField } from "../../../../../../components/ha-textfield";
+
+import "../../../../../../components/ha-textfield";
+import "../../../../../../components/ha-alert";
+
+@customElement("zwave-js-add-node-code-input")
+export class ZWaveJsAddNodeCodeInput extends LitElement {
+  @property() public value = "";
+
+  @property() public description = "";
+
+  @property() public placeholder = "";
+
+  @property({ attribute: "reference-key" }) public referenceKey = "";
+
+  @property() public error?: string;
+
+  @property({ type: Boolean }) public numeric = false;
+
+  render() {
+    return html`
+      <p>${this.description}</p>
+      ${this.error
+        ? html`<ha-alert alert-type="error">${this.error}</ha-alert>`
+        : nothing}
+      <ha-textfield
+        .placeholder=${this.placeholder}
+        .value=${this.value}
+        @input=${this._handleChange}
+        @keyup=${this._handleKeyup}
+        required
```

Comment:
Should the type if the textfield also be set to numeric if `this.numeric === true`?

Reply:
No, because the code can start with leading 0